### PR TITLE
Added Geometry.computeFlatVertexNormals()

### PR DESCRIPTION
**Pull request by WestLangley**
Also added `computeFaceNormals()` inside `computeVertexNormals()` since face normals are required for the algorithm to work.

### CODE REVIEW

File: src/core/Geometry.js [hunk 1]
```diff
@@ -399,6 +399,8 @@ Object.assign( THREE.Geometry.prototype, THREE.EventDispatcher.prototype, {
 
 	computeVertexNormals: function ( areaWeighted ) {
 
+		this.computeFaceNormals();
```
Comment (mrdoob) [inline src/core/Geometry.js]: Was this change intentional?

Comment (WestLangley) [inline src/core/Geometry.js]: Yes. Since face normals are required for the computation, I decided to add that line to both this method and the new `ComputeFlatVertexNormals()`. It was a judgement call.

Hmmm. Now I see normals are only required for the non-area-weighted case...

**Comment by Astrak** [comment]
Thanks for that method. I closed #9217 and reference the original issue in this one : #7130

**Comment by spacejack** [comment]
Great, I just realized I needed this feature! Thanks!

File: src/core/Geometry.js (hunk 1 above)
Comment (WestLangley) [inline src/core/Geometry.js]: @mrdoob What is your preference? Do you want to leave `computeVertexNormals()` as is?

Comment (WestLangley) [inline src/core/Geometry.js]: @mrdoob OK. Changed... Acceptable to you?

**Comment by WestLangley** [comment]
@mrdoob How do you feel about this?

**Comment by gamedolphin** [comment]
This helped out a lot. Thank you. :)

**Comment by mrdoob** [comment]
Thanks!
